services/nexus/nexus.service.js

```diff
@@ -1,130 +1,230 @@
+    return { color: 'blue', label: 'nexus' }
+  }
+
   static get examples() {
     return [
       {
         title: 'Sonatype Nexus (Releases)',
-        previewUrl: 'r/https/oss.sonatype.org/com.google.guava/guava',
+        pattern: 'r/:scheme/:host/:groupId/:artifactId',
+        namedParams: {
+          scheme: 'https',
+          host: 'oss.sonatype.org',
+          groupId: 'com.google.guava',
+          artifactId: 'guava',
+        },
+        staticExample: this.render({
+          version: 'v27.0.1-jre',
+        }),
       },
       {
         title: 'Sonatype Nexus (Snapshots)',
-        previewUrl: 's/https/oss.sonatype.org/com.google.guava/guava',
+        pattern: 's/:scheme/:host/:groupId/:artifactId',
+        namedParams: {
+          scheme: 'https',
+          host: 'oss.sonatype.org',
+          groupId: 'com.google.guava',
+          artifactId: 'guava',
+        },
+        staticExample: this.render({
+          version: 'v24.0-SNAPSHOT',
+        }),
+      },
+      {
+        title: 'Sonatype Nexus (Repository)',
+        pattern: ':repo/:scheme/:host/:groupId/:artifactId',
+        namedParams: {
+          repo: 'developer',
+          scheme: 'https',
+          host: 'repository.jboss.org/nexus',
+          groupId: 'ai.h2o',
+          artifactId: 'h2o-automl',
+        },
+        staticExample: this.render({
+          version: '3.22.0.2',
+        }),
+      },
+      {
+        title: 'Sonatype Nexus (Query Options)',
+        pattern: ':repo/:scheme/:host/:groupId/:artifactId/:queryOpt',
+        namedParams: {
+          repo: 'fs-public-snapshots',
+          scheme: 'https',
+          host: 'repository.jboss.org/nexus',
+          groupId: 'com.progress.fuse',
+          artifactId: 'fusehq',
+          queryOpt: ':c=agent-apple-osx:p=tar.gz',
+        },
+        staticExample: this.render({
+          version: '7.0.1-SNAPSHOT',
+        }),
+        documentation: `
+        <p>
+          Note that you can use query options with any Nexus badge type (Releases, Snapshots, or Repository)
+        </p>
+        <p>
+          Query options should be provided as key=value pairs separated by a semicolon
+        </p>
+        `,
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    // standalone sonatype nexus installation
+  transform({ repo, json }) {
+    let version
+    if (repo === 'r') {
+      version = json.data[0].latestRelease
+    } else if (repo === 's') {
+      for (const artifact of json.data) {
+        if (isSnapshotVersion(artifact.latestSnapshot)) {
+          version = artifact.latestSnapshot
+          break
+        }
+        if (isSnapshotVersion(artifact.version)) {
+          version = artifact.version
+          break
+        }
+      }
+    } else {
+      version = json.data.baseVersion || json.data.version
+    }
+
+    if (!version) {
+      throw new InvalidResponse({ prettyMessage: 'invalid artifact version' })
```

Comment:
It's not necessary to change this, though if you move this bit back to `handle()` you can use `return` statements above.